moshi/src/main/java/com/squareup/moshi/internal/Util.kt

```diff
@@ -61,11 +62,7 @@ private val METADATA: Class<out Annotation>? = try {
 // We look up the constructor marker separately because Metadata might be (justifiably)
 // stripped by R8/Proguard but the DefaultConstructorMarker is still present.
 @JvmField
-public val DEFAULT_CONSTRUCTOR_MARKER: Class<*>? = try {
-  Class.forName("kotlin.jvm.internal.DefaultConstructorMarker")
-} catch (ignored: ClassNotFoundException) {
-  null
-}
+public val DEFAULT_CONSTRUCTOR_MARKER: Class<*> = DefaultConstructorMarker::class.java
```

Comment:
Let's make this work like the `METADATA` field. We do this a particular way to prevent R8 from automatically keeping it

Reply:
Using reflection means we have to keep all(This class is just a marker, it only has a name and no content.), using references allows R8 to obfuscate safely.